resolver/src/lookup.rs

```diff
@@ -35,19 +38,41 @@ use resolver_future::BasicResolverHandle;
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Lookup {
     rdatas: Arc<Vec<RData>>,
+    valid_until: Instant,
 }
 
 impl Lookup {
-    /// Return new instance with given rdatas
-    pub fn new(rdatas: Arc<Vec<RData>>) -> Self {
-        Lookup { rdatas }
+    /// Return new instance with given rdatas and the maximum TTL.
+    pub fn new(rdatas: Arc<Vec<RData>>,) -> Self {
+        let valid_until =
+            Instant::now() + Duration::from_secs(MAX_TTL as u64);
```

Comment:
Is there an `Instant::MAX` or something like it that can be used instead? 
`Instant::now()` is relatively expensive (potentially a syscall). `MAX_TTL` is for parsing/validating input on the wire.

If there's no `Instant::MAX` or equivalent then maybe we should use something like `enum Validity { ValidUntil(Instant), Forever }`, isomorphic to `Option`, with comparison semantics such that `ValidUntil(x) < Forever` for all x.

Reply:
@briansmith 
> If there's no `Instant::MAX` or equivalent then maybe we should use something like `enum Validity { ValidUntil(Instant), Forever }`, isomorphic to `Option`, with comparison semantics such that `ValidUntil(x) < Forever` for all x.

I like this approach (and you're correct that there's no `Instant::MAX`). 

However, I would like to get @bluejekyll's opinion on whether we should do something like this rather than using MAX_TTL` as he suggested in https://github.com/bluejekyll/trust-dns/pull/444#discussion_r185392627 before I make more changes.

Reply:
Personally I’m fine with MAX_TTL. I understand the concern @briansmith has, but I don’t feel like we need to over complicate this. As long as we guarantee that the correct TTL is used on insert.

Forcing the caller to pick a good value, rather than a decent default, seems potentially error prone as well. We could change ‘new’ to ‘with_max_ttl’ to make it more explicit.